apps/app/src/utils/pageScroll.ts

```diff
@@ -1,13 +1,11 @@
 import "client-only";
 
-const classes = ["overflow-hidden"];
+const classes = ["overflow-hidden", "sm:overflow-auto"];
 
 export const lockScroll = () => {
-	document.body.style.paddingRight = `${window.innerWidth - document.body.offsetWidth}px`;
 	document.body.classList.add(...classes);
 };
 
 export const unlockScroll = () => {
-	document.body.style.paddingRight = "";
```

Comment:
Dlaczego to usunąłeś?

Reply:
@mmiszy, poniżej pokazuję, jak działa strona ze skasowanymi linijkami kodu.
Zakładałem scenariusz:
1. otwieram stronę i otwieram menu:

Wszystko jest ok.
![1](https://user-images.githubusercontent.com/27455716/210146277-92616c26-423f-4166-a41f-e9d51807ce6e.png)

2. przełączam stroną w tryb horyzontalny:

Po prawej stronie zostaje dodatkowa przerwa między zawartością strony, a paskiem przewijania.
![2](https://user-images.githubusercontent.com/27455716/210146892-5c60e6b6-301f-4557-8426-1cf8b838cd79.png)

Reply:
Nie możesz po prostu kasować kodu, jeśli nie rozumiesz co on robi...

Reply:
masz rację, ale teraz wydawało mi się, że pozostali kontrybutorzy mają wolne / przerwę i nie chciałem przeszkadzać, a powinienem dopytać @AdiPol1359

Reply:
@AdiPol1359, jeśli możesz wskaż dokładnie w jakim przypadku przydaje się funkcjonalność z tej linijki kodu:
https://github.com/typeofweb/devfaq/blob/3f66263c79c3db8b75e76e1819ac1f2146785ac8/apps/app/src/utils/pageScroll.ts#L6

znaczy domyślam się co rozwiązuje, ale wydaje mi się zbędna / przeszkadza w rozwiązaniu trochę innego problemu.

Reply:
Czy jest to potrzebne tylko w przypadku kiedy ukrywamy scrollbar i wyświetlamy modal, a dodatkowy padding dodawany do `body` ma za zadanie zapobiegać przesunięciom strony podczas otwierania / zamykania modala?

Reply:
> @AdiPol1359, jeśli możesz wskaż dokładnie w jakim przypadku przydaje się funkcjonalność z tej linijki kodu:
> 
> https://github.com/typeofweb/devfaq/blob/3f66263c79c3db8b75e76e1819ac1f2146785ac8/apps/app/src/utils/pageScroll.ts#L6
> 
> znaczy domyślam się co rozwiązuje, ale wydaje mi się zbędna / przeszkadza w rozwiązaniu trochę innego problemu.

Tak jak napisałeś, kod ma zapobiegać przesuwaniu się strony podczas otwierania modala. Podczas chowania scrolla obliczany jest padding, który należy dodać żeby uniknąć przesunięcia.

Reply:
Warto zwrócić uwagę, że ten padding różni się w zależności od systemu operacyjnego i jego ustawień, a także przeglądarki. Nie da się tego zahardkodować w CSS.

Reply:
Ok. rozumiem już o co chodzi. Większy mam teraz problem z takim scenariuszem:
1. otwieram stronę na smartfonie w trybie pionowym,
2. otwieram menu
3. obracam smartfon, aby wyświetlać treść w trybie 'poziomym' - chowa mi się menu, bo dodałem `sm:overflow-auto`
4. otwieram modal np. 'Dodaj pytanie' i okazuje się, że można przewijać stronę, bo znacznik `body` nadal ma klasę `sm:overflow-auto`.
Zastanawiam się czy próbować rozwiązać ten problem, czy uznać to za naprawdę rzadki przypadek brzegowy i sobie go odpuścić, bo ciężko wymyśleć mi jakieś w miarę czytelne rozwiązanie do tego.